pkg/repository/repository.go

```diff
@@ -46,12 +47,48 @@ func Parse(s string) (Repository, error) {
 	case 3:
 		return irepo.New(parts[0], parts[1], parts[2]), nil
 	case 2:
-		host := os.Getenv("GH_HOST")
-		if host == "" {
-			host = "github.com"
+		host := "github.com"
+		cfg, err := config.Load()
```

Comment:
This seems correct for now, but ultimately I find it surprising that `Parse()` implicitly loads config. I think an ideal API might be one where a repository "parser" is derived from a config object, so that their relationship is denoted explicitly.

Reply:
Yeah I agree it is a bit odd. `go-gh` is also targeting extension authors, I would think that they don't necessarily want to concern themselves with loading up the config and determining the correct host. We also do this when initializing the various API clients. As long as there are alternatives for advanced users, that don't want automatic resolution such as `repository.ParseWithHost`, I am okay with having these easy to use alternatives.